### DISCUSSION TITLE
Explanation of GHSA-2f5v-8r3f-8pww

### DISCUSSION THREAD

**Discussion by hintofbasil** [argoproj/argo-cd#8889] [Q&A]
There was a security advisory posted yesterday (23rd March 2022) around privilege escalation.  https://github.com/argoproj/argo-cd/security/advisories/GHSA-2f5v-8r3f-8pww

The current information surrounding this does not go in to enough detail to understand how this would be exploited and therefore it's very difficult to understand the potential impact of this based on our current configuration.

My current understanding from reading [the fix PR](https://github.com/argoproj/argo-cd/commit/af03b291d4b7e9d3ce9a6580ae9c8141af0e05cf) is that the project based resource deny list, allowed repositories, etc was not being applied to children of deployed resources.  This allowed a malicious actor with permissions to change an applications resources to declare an existing resource as a child of a resource and so deploy, delete, or view that child resource.

I can not seem to find a way to have ArgoCD consider an arbitrary resource a child of a currently deployed resource and so can't make this exploit work.

If someone has a better understanding of the issue could shed some light on exactly what the issue was here that would be highly appreciated.

**Comment by alexmt** [comment]
Hello @hintofbasil ,

API relies on data reported by the controller. The controller stores resource identifier in the resource tree only if it is permitted by the project, that is why the check is missing in API server itself. The controller check is here:

https://github.com/argoproj/argo-cd/blob/d7608ecc44f873c5992189735d4eaf27feb995c8/controller/appcontroller.go#L426

**Comment by c1505** [comment]
I apologize if this is considered hijacking the discussion, but I am also a bit confused with the stated affected versions being 0.5.0 through 2.1.12, 2.2.7, 2.3.1 , but the section on the impact speaks of impact with versions starting with v1.0.0, v0.8.0, and v0.5.0 .  I assume there is impact for the versions stated as affected.  Is it my misreading of the impact section or is it unknown what the impact is for v2.x.x ?

**Reply by crenshaw-dev** [reply]
No worries! Sorry that was unclear. Every version starting with v1.0.0 (including all 2.x series) before the patched versions is vulnerable to all the listed exploits.

0.8.0 <= version < 1.0.0 are vulnerable to a subset of the exploits.
0.5.0 <= version < 0.8.0 are vulnerable to an even smaller subset of the exploits.

**Reply by c1505** [reply]
Thanks for the clarification!
